lib/elixir/src/elixir_erl_var.erl

```diff
@@ -114,7 +114,8 @@ should_warn(Meta) ->
 mergev(S1, S2) ->
   S2#elixir_erl{
     vars=merge_vars(S1#elixir_erl.vars, S2#elixir_erl.vars),
-    export_vars=merge_opt_vars(S1#elixir_erl.export_vars, S2#elixir_erl.export_vars)
+    export_vars=merge_opt_vars(S1#elixir_erl.export_vars, S2#elixir_erl.export_vars),
+    ssa_types=maps:merge(S1#elixir_erl.ssa_types, S2#elixir_erl.ssa_types)
```

Comment:
@michalmuskala cool. I will shelve this PR for now because it is not worth adding the complexity. Some things will be announced during ElixirConf EU that may shine some light on this but it is probably best to clean up the compiler before adding this complexity. I will list it under the previously mentioned issue. Thank you!